File: Mantra-Aur_audio_automation/shivjiAutomation/create_tracks.js
const fs = require('fs');
const https = require('https');
const path = require('path');

// --- CONFIGURATION ---
const ACCESS_TOKEN = "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE3NzY1MDQ0NTYsInVzZXJfbmFtZSI6IjEiLCJhdXRob3JpdGllcyI6WyJST0xFX0RJVklORV9ERVZJQ0VfU1VQUE9SVCIsIlJPTEVfRERBUFBfVVNFUiIsIlJPTEVfRERfUFVKQV9DQVNISUVSIiwiUk9MRV9ERF9QVUpBX01BTkFHRVIiLCJST0xFX0RJVklORV9URU1QTEVfVklFV19NQU5BR0VSIiwiUk9MRV9ESVZJTkVfRlJBTkNISVNFRV9DQVNISUVSIiwiUk9MRV9ERF9QVUpBX0FETUlOIiwiUk9MRV9ERF9LSVRfQkFDS09GRklDRSIsIlJPTEVfRERBUFBfQURNSU4iLCJST0xFX0RJVklORV9GUkFOQ0hJU0VFX09QRVJBVE9SIiwiUk9MRV9ESVZJTkVfQURNSU4iLCJST0xFX0RJVklORV9GUkFOQ0hJU0VFX0FDQ09VTlRBTlQiLCJST0xFX0REQVBQX1NVUFBPUlQiLCJST0xFX0RJVklORV9GUkFOQ0hJU0VFX01BTkFHRVIiLCJST0xFX0RJVklORV9GUkFOQ0hJU0VFIiwiUk9MRV9EREFQUF9DT05URU5UX01BTkFHRVIiLCJST0xFX0REX1BVSkFfTU9ERVJBVE9SIl0sImp0aSI6IlQtV3A5Ul9PWVR6eGVCREZvU2RPX1pQWnJzWSIsImNsaWVudF9pZCI6InRlY2h4ciIsInNjb3BlIjpbImFsbCJdfQ.qdDHJrZLvjhqDgTw9CtoKJAHZ8c_FFtNcq5f5_EcpQKPi3J79hRUfdKdFh1u10p-i05J7-ml7gEF382Tl5uFnE6OGhiN9SgML1XCFFJ01Z-Q91w_KWCHAG8hHo2Jeo6jH3cb_I9TH1A5nycWEoL8ZsscNGw7rNliWxE6cp9mNdH2IlPR6pY0w8BSsQQDsOuq7k3pkDH1iDadWjjoG69pF7su-GtWIyhfUlz7638JCrth-YQcfhFc7epaOxxywSVfm13tiyJ5x3fyBCf5q9rmnE9siVwIkRbSJo9r06TIZZNTGrM6pZ0IU4Hag-hxTBqvRpywQONhIB8SPMc1Lo-0ww"; // Replace with your actual token
const HOST = "k8uatgateway.techxrdev.in"; // Replace with the actual environment URL if needed
const INPUT_FILE = path.join(__dirname, 'response.json');

// Set DRY_RUN=true to preview requests without creating.
const DRY_RUN = true;

// Set LIMIT to a number (e.g. 5) to create only a few tracks.
// Set LIMIT=null to process all tracks.
const LIMIT = null;

// Optional pagination control when running in chunks.
const START_INDEX = 0;

const REQUEST_DELAY_MS = 1000;
// ---------------------

function createTrackRequest(payload) {
    return new Promise((resolve, reject) => {
        const data = JSON.stringify(payload);
        const options = {
            hostname: HOST,
            path: `/api/content/shubhdarshan/v1/add/track`,
            method: 'POST',
            headers: {
                'Content-Type': 'application/json',
                'Authorization': `Bearer ${ACCESS_TOKEN}`,
                'Content-Length': Buffer.byteLength(data)
            }
        };

        if (DRY_RUN) {
            resolve({
                status: 200,
                body: 'Dry run success',
                endpoint: `https://${HOST}${options.path}`,
                payload
            });
            return;
        }

        const req = https.request(options, (res) => {
            let body = '';
            res.on('data', (chunk) => {
                body += chunk;
            });
            res.on('end', () => {
                if (res.statusCode >= 200 && res.statusCode < 300) {
                    resolve({ status: res.statusCode, body });
                } else {
                    reject({ status: res.statusCode, body });
                }
            });
        });

        req.on('error', (e) => reject(e));
        req.write(data);
        req.end();
    });
}

async function main() {
    try {
        const allTracks = JSON.parse(fs.readFileSync(INPUT_FILE, 'utf8'));
        if (!Array.isArray(allTracks)) {
            throw new Error('Input file must contain a JSON array of track payloads.');
        }

        const sliced = allTracks.slice(START_INDEX);
        const tracksToProcess = LIMIT === null ? sliced : sliced.slice(0, LIMIT);

        console.log(`Input Payloads: ${allTracks.length}`);
        console.log(`START_INDEX: ${START_INDEX}`);
        console.log(`LIMIT: ${LIMIT === null ? 'ALL' : LIMIT}`);
        console.log(`Tracks to process: ${tracksToProcess.length}`);
        console.log(`DRY_RUN: ${DRY_RUN}`);

        let successCount = 0;
        let failCount = 0;

        for (let i = 0; i < tracksToProcess.length; i++) {
            const payload = tracksToProcess[i];
            const title = payload?.thumbnail?.title?.en || payload?.thumbnail?.title?.hi || 'N/A';
            const endpoint = `https://${HOST}/api/content/shubhdarshan/v1/add/track`;

            console.log(`\n[${i + 1}/${tracksToProcess.length}] ${title}`);
            console.log(`Endpoint: POST ${endpoint}`);

            try {
                if (DRY_RUN) {
                    console.log('[DRY RUN] Payload to be sent:');
                    console.log(JSON.stringify(payload, null, 2));
                }

                const result = await createTrackRequest(payload);
                console.log(`Success. Status: ${result.status}`);
                if (!DRY_RUN && result.body) {
                    try {
                        const parsed = JSON.parse(result.body);
                        console.log(`Response ID: ${parsed.id}`);
                    } catch (e) {
                        console.log(`Response: ${result.body}`);
                    }
                }
                successCount++;
            } catch (error) {
                failCount++;
                console.error(`Failed. Status: ${error.status || 'Error'}`);
                console.error(`Error Body: ${error.body || error.message}`);
            }

            if (!DRY_RUN) {
                await new Promise((resolve) => setTimeout(resolve, REQUEST_DELAY_MS));
            }
        }

        console.log('\n--- SUMMARY ---');
        console.log(`Total selected: ${tracksToProcess.length}`);
        console.log(`Succeeded: ${successCount}`);
        console.log(`Failed: ${failCount}`);
    } catch (error) {
        console.error(`Critical Error: ${error.message}`);
        process.exit(1);
    }
}

main();

File: Mantra-Aur_audio_automation/ApiDataPopulation/create_tracks.py
# import json
# import http.client
# import time
# import os
# import ssl

# # --- CONFIGURATION ---
# ACCESS_TOKEN = "eyJhbGciOiJSUzI1NiIsInR5cCI6IkpXVCJ9.eyJleHAiOjE3Nzc3MjQxMjQsInVzZXJfbmFtZSI6IjEiLCJhdXRob3JpdGllcyI6WyJST0xFX0REQVBQX1VTRVIiLCJST0xFX0RJVklORV9ERVZJQ0VfU1VQUE9SVCIsIlJPTEVfRERfUFVKQV9DQVNISUVSIiwiUk9MRV9ERF9QVUpBX01BTkFHRVIiLCJST0xFX0RJVklORV9GUkFOQ0hJU0VFX0NBU0hJRVIiLCJST0xFX0RJVklORV9URU1QTEVfVklFV19NQU5BR0VSIiwiUk9MRV9ERF9LSVRfQkFDS09GRklDRSIsIlJPTEVfRERfUFVKQV9BRE1JTiIsIlJPTEVfRERBUFBfQURNSU4iLCJST0xFX0RJVklORV9BRE1JTiIsIlJPTEVfRElWSU5FX0ZSQU5DSElTRUVfT1BFUkFUT1IiLCJST0xFX0REQVBQX1NVUFBPUlQiLCJST0xFX0RJVklORV9GUkFOQ0hJU0VFX0FDQ09VTlRBTlQiLCJST0xFX0RJVklORV9GUkFOQ0hJU0VFX01BTkFHRVIiLCJST0xFX0RJVklORV9GUkFOQ0hJU0VFIiwiUk9MRV9EREFQUF9DT05URU5UX01BTkFHRVIiLCJST0xFX0REX1BVSkFfTU9ERVJBVE9SIl0sImp0aSI6InlHOVBHUFBwdmptQWl4Y2ExZU1adFR5Ty1NTSIsImNsaWVudF9pZCI6InRlY2h4ciIsInNjb3BlIjpbImFsbCJdfQ.JUJR082yjBRA8d0uNPcuGauJELbCWInQItpmDwUxhCH4g7Tw4QTR5NQg9ajhbw5F6IbOa7sfwexKCdgojI17lXYFu9tOkdSQ4QnBT28fBJC8S83A-vyFoPCSAkNzrRfiKfWHyJtvsLGmT6wyVC9_J0bEYdiszUNPxfi9j3uJ1z4FN-XLKgIOR9Qm2OHrPdjA4-fUfPsAeOjHqJxnPKg3__PZJFFq1RHIo1lJT-_2wVy2zIUUwu8jUbci9GRjvbLJyvnC8JFgGFyO0x-90iecBqhhPOiXh0NrvEYnTmc9_OvQYsLYLWmCOw9nhSPeTeptDzKJTZSKt_bHT9Vq1vZVxQ"
# HOST = "k8uatgateway.techxrdev.in"
# INPUT_FILE = "CleanTracksDTO.json"

# # Set DRY_RUN=True to preview requests without sending.
# DRY_RUN = False

# # Set LIMIT to a number (e.g. 5) to update only a few tracks.
# # Set LIMIT=None to process all tracks.
# LIMIT = None

# # Optional pagination control when running in chunks.
# START_INDEX = 0

# REQUEST_DELAY_SECONDS = 1
# # ---------------------

# def create_track_request(conn, payload):
#     headers = {
#         'Content-Type': 'application/json',
#         'Authorization': f'Bearer {ACCESS_TOKEN}'
#     }
#     data = json.dumps(payload)
    
#     endpoint = "/api/content/shubhdarshan/v1/admin/tracks"
    
#     if DRY_RUN:
#         return {
#             "status": 200,
#             "body": "Dry run success",
#             "endpoint": f"https://{HOST}{endpoint}",
#             "payload": payload
#         }

#     try:
#         conn.request("POST", endpoint, body=data, headers=headers)
#         res = conn.getresponse()
#         res_data = res.read().decode("utf-8")
        
#         if 200 <= res.status < 300:
#             return {"status": res.status, "body": res_data}
#         else:
#             return {"status": res.status, "body": res_data, "error": True}
#     except Exception as e:
#         return {"status": "Error", "body": str(e), "error": True}

# def main():
#     try:
#         # Resolve path relative to script directory
#         script_dir = os.path.dirname(os.path.abspath(__file__))
#         file_path = os.path.join(script_dir, INPUT_FILE)
        
#         if not os.path.exists(file_path):
#             print(f"Error: Input file not found at {file_path}")
#             return

#         with open(file_path, 'r', encoding='utf-8') as f:
#             all_tracks = json.load(f)

#         if not isinstance(all_tracks, list):
#             print("Error: Input file must contain a JSON array of track payloads.")
#             return

#         # Slicing and Limiting
#         sliced = all_tracks[START_INDEX:]
#         tracks_to_process = sliced[:LIMIT] if LIMIT is not None else sliced

#         print(f"Input Payloads: {len(all_tracks)}")
#         print(f"START_INDEX: {START_INDEX}")
#         print(f"LIMIT: {'ALL' if LIMIT is None else LIMIT}")
#         print(f"Tracks to process: {len(tracks_to_process)}")
#         print(f"DRY_RUN: {DRY_RUN}")

#         success_count = 0
#         fail_count = 0

#         # Create HTTPS connection
#         context = ssl.create_default_context()
#         conn = http.client.HTTPSConnection(HOST, context=context)

#         for i, payload in enumerate(tracks_to_process):
#             title_en = payload.get('thumbnail', {}).get('title', {}).get('en', 'N/A')
#             tags = payload.get('tags', [])
            
#             print(f"\n[{i + 1}/{len(tracks_to_process)}] {title_en} | Tags: {tags}")
#             print(f"Endpoint: POST https://{HOST}/api/content/shubhdarshan/v1/admin/tracks")

#             if DRY_RUN:
#                 print("[DRY RUN] Payload snippet:")
#                 # Show only first bit of payload for brevity in terminal
#                 print(json.dumps(payload, indent=2, ensure_ascii=False))

#             result = create_track_request(conn, payload)

#             if "error" in result:
#                 fail_count += 1
#                 print(f"Failed. Status: {result['status']}")
#                 print(f"Error Body: {result['body']}")
#             else:
#                 success_count += 1
#                 print(f"Success. Status: {result['status']}")
#                 if not DRY_RUN and result.get('body'):
#                     try:
#                         parsed = json.loads(result['body'])
#                         print(f"Response: {json.dumps(parsed, indent=2)}")
#                     except:
#                         print(f"Response: {result['body']}")

#             if not DRY_RUN and i < len(tracks_to_process) - 1:
#                 time.sleep(REQUEST_DELAY_SECONDS)

#         conn.close()

#         print('\n--- SUMMARY ---')
#         print(f"Total selected: {len(tracks_to_process)}")
#         print(f"Succeeded: {success_count}")
#         print(f"Failed: {fail_count}")

#     except Exception as e:
#         print(f"Critical Error: {str(e)}")

# if __name__ == "__main__":
#     main()
